[ADD] module FDS Postfinance

ecino opened the pull request:
Superseeds #160
## Modules:

l10n_ch_fds_postfinance (allow users to import files from FDS PostFinance and convert imported files to bank statment)
l10n_ch_fds_upload_dd (allow users to upload direct debit order files to FDS PostFinance)

l10n_ch_fds_upload_sepa (allow users to upload sepa files to FDS PostFinance)
## Postfinance File Delivery Services

The file delivery services (FDS) is service offered by Postfinance AG Technology unit service.

FDS acts as a gateway between external networks and the Post CH SA. It enables mutual exchange of files between partners and Post CH applications.

Commit:
Add new modules for managing Postfinance fds files

ecino committed:
Code corrections

- Add missing security files
- Improve field names and strings

ecino committed:
Add french translation for fds_postfinance

yvaucher commented on l10n_ch_fds_upload_dd/wizard/fds_inherit_post_dd_export_upload_wizard.py:
```diff
@@ -0,0 +1,271 @@
+
+
+class fds_inherit_post_dd_export_upload_wizard(models.TransientModel):
+    ''' This addon allows you to upload the Direct Debit geberate file to
```
generated ?

yvaucher commented on l10n_ch_fds_upload_dd/wizard/fds_inherit_post_dd_export_upload_wizard.py:
```diff
@@ -0,0 +1,271 @@
+
+            :returns: None
+        '''
+        self.ensure_one()
```
No need for ensure_one, this method will work with multiple records

yvaucher commented on l10n_ch_fds_upload_sepa/README.md:
```diff
@@ -0,0 +1,47 @@
+------------
+
+* Nicolas Tran <nox.tran@gmail.com>
+* Please Compassion put your email here.
```
To fix

yvaucher commented on l10n_ch_fds_upload_sepa/__openerp__.py:
```diff
@@ -0,0 +1,46 @@
+{
+    'name': 'Swiss Postfinance FDS SEPA upload',
+    'summary': '''upload sepa files to FDS PostFinance''',
+    'version': '1.0',
```
version to set to 8.0.1.0

yvaucher commented on l10n_ch_fds_upload_sepa/model/fds_postfinance_account_sepa.py:
```diff
@@ -0,0 +1,43 @@
+    upload_sepa_directory = fields.Many2one(
+        comodel_name='fds.postfinance.files.directory',
+        string='default upload directory',
+        help='select one upload default directory.' +
```
The `+` is useless

Please use capital letter for first letter of each sentence

yvaucher commented on l10n_ch_fds_upload_sepa/model/fds_postfinance_account_sepa.py:
```diff
@@ -0,0 +1,43 @@
+        inverse_name='fds_account_id',
+        string='historical upload sepa',
+        readonly=True,
+        help='historical upload sepa'
```
This help text is not helpful

yvaucher commented on l10n_ch_fds_upload_sepa/view/fds_inherit_sepa_wizard_view.xml:
```diff
@@ -0,0 +1,42 @@
+	<record id="wizard_pain_001_view_inherited" model="ir.ui.view">
+		<field name="name">Create SEPA Payment - Upload extension</field>
+		<field name="model">wizard.pain001</field>
+        <field name="inherit_id" ref="l10n_ch_sepa.wizard_pain_001_view"/>
```
xml indentation is a bit messy

yvaucher commented on l10n_ch_fds_upload_sepa/wizard/fds_inherit_sepa_wizard.py:
```diff
@@ -0,0 +1,319 @@
+        self.create_pain_001()
+        attachment_obj = self.env['ir.attachment']
+        payment_order_id = self.env.context.get('active_id')
+        file = attachment_obj.search([['res_id', '=', payment_order_id]])[0]
```
Please rename that variable file is a reserved term

yvaucher replied:
You can add `limit=1` to your `search()`

yvaucher commented on l10n_ch_fds_upload_sepa/wizard/fds_inherit_sepa_wizard.py:
```diff
@@ -0,0 +1,319 @@
+
+        # check if only one fds account
+        existing_account_ids = self.fds_account.search([]).ids
+        if len(existing_account_ids) != 1:
```
you can use `search_count([])`

yvaucher commented on l10n_ch_fds_upload_sepa/wizard/fds_inherit_sepa_wizard.py:
```diff
@@ -0,0 +1,319 @@
+
+        # check if default upload directory is allowed
+        dir_name = self.fds_account.upload_dd_directory.name
+        dir = self.fds_account_directory.search([('name', '=', dir_name)])
```
Please don't use built-in term `dir`

yvaucher commented on l10n_ch_fds_upload_sepa/wizard/fds_inherit_sepa_wizard.py:
```diff
@@ -0,0 +1,319 @@
+            tmp_key = self._create_tmp_file(key.private_key_crypted, tmp_d)[0]
+            tmp_f = self._create_tmp_file(self.attachment_id.datas, tmp_d)[0]
+            old_path_f = os.path.join(tmp_d, tmp_f.name)
+            new_path_f = os.path.join(tmp_d, ''.join(self.filename.split(':')))
```
`''.join(self.filename.split(':'))` could be `self.filename.replace(':', '')`

yvaucher commented on l10n_ch_fds_upload_sepa/wizard/fds_inherit_sepa_wizard.py:
```diff
@@ -0,0 +1,319 @@
+        '''
+        fds_authentication_key_obj = self.env['fds.authentication.keys']
+        key = fds_authentication_key_obj.search([
+            ['user_id', '=', self.env.user.id],
```
you can use `self.env.uid`

yvaucher commented on l10n_ch_fds_upload_sepa/wizard/fds_inherit_sepa_wizard.py:
```diff
@@ -0,0 +1,319 @@
+            ['fds_account_id', '=', self.fds_account.id]])
+
+        if not key:
+            raise exceptions.Warning('You don\'t have key')
```
You could use double quote for all error message to avoid this kind of escaping.

yvaucher commented on oca_dependencies.txt:
```diff
@@ -3,5 +3,6 @@
 bank-statement-reconcile
-bank-statement-import
\ No newline at end of file
+https://github.com/OCA/bank-statement-import.git
```
Any reason why bank-statement-import is there twice ?

yvaucher commented on l10n_ch_fds_upload_sepa/wizard/fds_inherit_sepa_wizard.py:
```diff
@@ -0,0 +1,319 @@
+    )
+    state = fields.Selection(
+        selection=[('default', 'Default'),
+                   ('finish', 'finish'),
```
It would be better with capital letter at the beginning of displayed word to the user.

yvaucher commented on l10n_ch_fds_upload_sepa/wizard/fds_inherit_sepa_wizard.py:
```diff
@@ -0,0 +1,319 @@
+        file = attachment_obj.search([['res_id', '=', payment_order_id]])[0]
+        self.write({'attachment_id': file.id})
+        self.write({'payment_order_id': payment_order_id})
+        self.write({'state': 'finish'})
```
Please do a single write with all values, it will do only one `UPDATE` in database instead of 3

yvaucher commented on l10n_ch_fds_upload_sepa/wizard/fds_inherit_sepa_wizard.py:
```diff
@@ -0,0 +1,319 @@
+        self.fds_account = existing_account_ids[0]
+
+        # check if default upload directory exist
+        if not self.fds_account.upload_sepa_directory:
```
Couldn't you include that check in the domain of the previous `search` ?

ecino replied:
previous search is not on the same model

yvaucher replied:
Correct me if I'm wrong, but the search is done on `self.fds_account`

Thus you assign your result to `self.fds_account`

And finally you check that `self.fds_account` has a `upload_sepa_directory`

Sounds to me you can write:

```
existing_account_ids = self.fds_account.search([('upload_sepa_directory', '=', 'True']).ids
if len(existing_account_ids) != 1:
    return self._do_populate_tasks()
self.fds_account = existing_account_ids[0]
```

Unless you search on 'fds.postfinance.account' model can return you multiple results and you want to call _do_poplate_tasks in that case.

Commit:
fix (whitespace, some English grammar, variable naming)

yvaucher commented on l10n_ch_fds_postfinance/model/fds_postfinance_account.py:
```diff
@@ -0,0 +1,238 @@
+    postfinance_email = fields.Char(
+        default='fds@post.ch',
+        required=True,
+        help='E-mail of fds postfinance'
```
s/fds/FDS

yvaucher commented on l10n_ch_fds_postfinance/model/fds_postfinance_account.py:
```diff
@@ -0,0 +1,238 @@
+    #         Button action          #
+    ##################################
+    @api.multi
+    def verifyDirectories_button(self):
```
Please don't mix camel case and snake case.

BTW: in python conventions are snake_case for function name and variable
CamelCase is used for class names

yvaucher commented on l10n_ch_fds_upload_sepa/wizard/fds_inherit_sepa_wizard.py:
```diff
@@ -0,0 +1,320 @@
+    ##############################
+    @api.multi
+    def _get_sftp_key(self):
+        ''' private function that get the sftp key need for connection
```
s/sftp need/SFTP needed

yvaucher commented on l10n_ch_fds_upload_sepa/wizard/fds_inherit_sepa_wizard.py:
```diff
@@ -0,0 +1,320 @@
+            :returns: None
+        '''
+        self.ensure_one()
+        self.write({'state': 'finish'})
```
As you ensure you have one record, you can write `self.state =`finish`otherwise you can remove`self.ensure_one()`

yvaucher commented on l10n_ch_fds_upload_sepa/model/fds_postfinance_historical_sepa.py:
```diff
@@ -0,0 +1,63 @@
+
+    fds_account_id = fields.Many2one(
+        comodel_name='fds.postfinance.account',
+        string='FDS account id',
```
We might prefer "FDS account" `id` is only there to say it's a relation, we don't want to display it to the user.

yvaucher commented on l10n_ch_fds_upload_sepa/model/fds_postfinance_historical_sepa.py:
```diff
@@ -0,0 +1,63 @@
+        string='payment order id',
+        ondelete='restrict',
+        readonly=True,
+        help='payment order id'
```
Payment order of what ?

yvaucher commented on l10n_ch_fds_upload_sepa/model/fds_postfinance_historical_sepa.py:
```diff
@@ -0,0 +1,63 @@
+                   ('uploaded', 'Uploaded')],
+        readonly=True,
+        default='not_uploaded',
+        help='state of file'
```
s/state of file/Upload state of the file/ ?

yvaucher commented on l10n_ch_fds_upload_sepa/wizard/fds_inherit_sepa_wizard.py:
```diff
@@ -0,0 +1,320 @@
+        comodel_name='ir.attachment',
+        string='File attachment',
+        readonly=True,
+        help='[info] keep one recored of the model attachment'
```
recored ?

yvaucher commented on l10n_ch_fds_upload_sepa/wizard/fds_inherit_sepa_wizard.py:
```diff
@@ -0,0 +1,320 @@
+        comodel_name='payment.order',
+        string='Payment order',
+        readonly=True,
+        help='payment order id'
```
not helpful

yvaucher commented on l10n_ch_fds_upload_sepa/README.md:
```diff
@@ -0,0 +1,50 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
```
Check the template, it has a bit evolved:

https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

yvaucher commented on l10n_ch_fds_upload_dd/model/fds_postfinance_historical_dd.py:
```diff
@@ -0,0 +1,63 @@
+    filename = fields.Char(
+        string='Filename',
+        readonly=True,
+        help='The name of the file'
```
What file ? What is in this file

yvaucher commented on l10n_ch_fds_postfinance/i18n/fr_CH.po:
```diff
@@ -0,0 +1,694 @@
+#. module: l10n_ch_fds_postfinance
+#: help:fds.postfinance.account,fds_postfinance_files_ids:0
+msgid "downloaded files from sftp"
+msgstr "Fichiers téléchargés du sftp"
```
sftp/SFTP

yvaucher commented on l10n_ch_fds_postfinance/i18n/fr_CH.po:
```diff
@@ -0,0 +1,694 @@
+"user that have create/contact with the fds postfinance account, it will be "
+"used to send the public key to postfinance"
+msgstr ""
+"Utilisateur qui a contact avec PostFinance. Sera utilisé pour envoyer le clé "
```
I don't understand this one neither do I understand the english texte

yvaucher commented on l10n_ch_fds_postfinance/i18n/fr_CH.po:
```diff
@@ -0,0 +1,694 @@
+#. module: l10n_ch_fds_postfinance
+#: model:ir.actions.act_window,name:l10n_ch_fds_postfinance.action_fds_files_import_tobankstatments_wizard
+msgid "FDS files import to bank Statments"
+msgstr "Import FDS à Relevé bancaires"
```
s/".*"/"Import des fichiers FDS dans des Relevés bancaires"/

yvaucher commented on l10n_ch_fds_postfinance/i18n/fr_CH.po:
```diff
@@ -0,0 +1,694 @@
+#. module: l10n_ch_fds_postfinance
+#: view:fds.postfinance.files:l10n_ch_fds_postfinance.fds_postfinance_files_form
+msgid "Reset to draft"
+msgstr "Remettre à brouillon"
```
s/à/en

ecino committed:
Rework of FDS modules

- Change class names and choose more comprehensive names
- Choose better field names
- Update and simplify the views
- Correct english mistakes
- Correct README files
- Correct translations

ecino commented:
I acknowledge these modules had some serious English issues leading to difficulties understanding the purpose of each object/field.

I tried to rename the objects and simplify the understanding of the module by updating descriptions and texts in the views.

I hope it is more clear now !

Thank you for the review

yvaucher commented:
@ecino Nice cleaning, that's much better and clear!

Just few nitpicking otherwise :+1:

ecino commented:
Done

yvaucher commented:
:+1:

ecino commented:
Do we need another approval for merging?

guewen commented:
:+1:
